botocore/httpchecksum.py

```diff
@@ -280,35 +290,30 @@ def resolve_request_checksum_algorithm(
             raise FlexibleChecksumError(
                 error_msg=f"Unsupported checksum algorithm: {algorithm_name}"
             )
+    elif request_checksum_required or (
+        algorithm_member and request_checksum_calculation == "when_supported"
+    ):
+        algorithm_name = "crc32"
+    else:
+        return
 
-        location_type = "header"
-        if operation_model.has_streaming_input:
-            # Operations with streaming input must support trailers.
-            if request["url"].startswith("https:"):
-                # We only support unsigned trailer checksums currently. As this
-                # disables payload signing we'll only use trailers over TLS.
-                location_type = "trailer"
-
-        algorithm = {
-            "algorithm": algorithm_name,
-            "in": location_type,
-            "name": f"x-amz-checksum-{algorithm_name}",
-        }
-
-        if algorithm["name"] in request["headers"]:
-            # If the header is already set by the customer, skip calculation
-            return
+    location_type = "header"
+    if operation_model.has_streaming_input:
+        # Operations with streaming input must support trailers.
+        if request["url"].startswith("https:"):
```

Comment:
Right, I was asking if this can be collapsed into:

```python
    if (
        operation_model.has_streaming_input
        and urlparse(request["url"]).scheme == "https"
    ): 
        # Operations with streaming input must support trailers.
```

or is there a reason we've made them distinct conditionals?

Reply:
Missed this comment yesterday. That makes much more sense, I misinterpreted your question. I updated this in 009ea3d3b3c07886c07863673d3dd471c8241647